src/RestaurantDetailPage.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+
+import { useSelector } from 'react-redux';
+
+import { get } from './utils';
+
+export default function RestaurantDetailPage() {
+  const restaurant = useSelector(get('restaurant'));
+
+  if (!restaurant.name) {
```

Comment:
레스토랑 이름이 없는 경우에 로딩 화면을 노출해주셨군요 👍🏼 

```suggestion
  if (!restaurant) {
```

이 경우엔 그냥 레스토랑이 없는 경우로 처리해줘도 좋겠네요 :)

Reply:
inititalState에서 retaurant를 빈 객체로 정의 해놨기 때문에, 
restaurant.name으로 처리해봤습니다 

빈 객체는 falsy가 아니기에 ! 로 처리하면 항상 falsy 가 되어 loading 문구가 나오지 않습니다. :) 

```
//reducer.js
const initialState = {
  regions: [],
  categories: [],
  restaurants: [],
  selectedRegion: null,
  selectedCategory: null,
  restaurant: {}, <- 빈 객체
};
```

Reply:
아하 빈 객체로 초기화 해주셨군요~ 제가 착각했네요 😅